compiler/resolution/lowerIterators.cpp

```diff
@@ -1919,6 +1907,16 @@ replaceErrorFormalWithEnclosingError(SymExpr* se) {
 // out of the enclosing loop. See also the PR message for #12963.
 //
 
+// 'bbcopy' may come from an IBB that simulates a throw and so have
+// a goto at the end. If so, remove the goto that we are inserting before.
+// Without this, multiple deinits may occur, ex.
+//   test/errhandling/parallel/forall-calls-throwing-fn2.chpl
+//
+static void adjustMultipleGotos(BlockStmt* bbcopy, GotoStmt* gt) {
+  if (isGotoStmt(bbcopy->body.tail))
```

Comment:
Should this only apply to a particular type of goto, like GOTO_BREAK_ERROR_HANDLING e.g.?

Reply:
I believe that duplicate gotos may cause an issue regardless of their kind. So I will keep it unconditional for now.